extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetMetadataFileLayout.java

```diff
@@ -196,15 +200,43 @@ public ParquetMetadataFileLayout(
                     partitions.put(partitionKey, partitionValue);
                 }
             }
-            final ParquetTableLocationKey tlk = new ParquetTableLocationKey(new File(directory, filePathString),
+            final File partitionFile = new File(directory, filePathString);
+            final ParquetTableLocationKey tlk = new ParquetTableLocationKey(partitionFile,
                     partitionOrder.getAndIncrement(), partitions, inputInstructions);
             tlk.setFileReader(metadataFileReader);
-            tlk.setMetadata(metadataFileMetadata);
+            tlk.setMetadata(getParquetMetadataForFile(partitionFile, metadataFileMetadata));
```

Comment:
I verified that that right metadata is being propagated here, but I don't know any tests which I can write for this, since column type is not partitioning, so not really reading any grouping data.